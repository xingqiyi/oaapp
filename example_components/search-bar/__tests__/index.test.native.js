import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import SearchBar from '../index';

describe('SearchBar', () => {
  it('renders correctly', () => {
    const wrapper = renderer.create(<SearchBar placeholder="搜索" />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  describe('test some events', () => {
    let handler;
    let wrapper;

    beforeEach(() => {
      handler = jest.fn();
    });

    it('fires onChange event', () => {
      wrapper = shallow(<SearchBar onChange={handler} />);
      wrapper.find('TextInput').simulate('changeText', 'foo');
      expect(handler).toBeCalledWith('foo');
    });

    it('fires onFocus event', () => {
      wrapper = shallow(<SearchBar onFocus={handler} />);
      wrapper.find('TextInput').simulate('focus');
      expect(handler).toBeCalledWith();
    });

    it('fires onBlur event', () => {
      wrapper = shallow(<SearchBar onBlur={handler} />);
      wrapper.find('TextInput').simulate('blur');
      expect(handler).toBeCalledWith();
    });

    it('fires onCancel event', () => {
      wrapper = shallow(<SearchBar value="test" showCancelButton onCancel={handler} />);
      wrapper.find('Text').simulate('press');
      expect(handler).toBeCalledWith('test');
    });
  });
});
