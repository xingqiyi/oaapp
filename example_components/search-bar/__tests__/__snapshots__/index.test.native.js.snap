// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#efeff4",
      "flexDirection": "row",
      "height": 44,
      "paddingLeft": 8,
      "paddingRight": 8,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <TextInput
      clearButtonMode="always"
      editable={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onClear={[Function]}
      onFocus={[Function]}
      onSubmit={[Function]}
      onSubmitEditing={[Function]}
      placeholder="搜索"
      prefixCls="am-search"
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderColor": "#dddddd",
            "borderRadius": 5,
            "borderWidth": 0.5,
            "color": "#000000",
            "flex": 1,
            "fontSize": 14,
            "height": 28,
            "paddingBottom": 0,
            "paddingLeft": 35,
            "paddingRight": 35,
            "paddingTop": 0,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <Image
    resizeMode="stretch"
    source={1}
    style={
      Object {
        "height": 15,
        "left": 16,
        "position": "absolute",
        "tintColor": "#cccccc",
        "top": 14.5,
        "width": 15,
      }
    }
  />
</View>
`;
